src/hawkbit-client.c

```diff
@@ -333,47 +337,55 @@ static gint rest_request(enum HTTPMethod method, const gchar* url, JsonBuilder*
                 token = g_strdup_printf("Authorization: GatewayToken %s",
                                         hawkbit_config->gateway_token);
         if (token)
-                if (!add_curl_header(&headers, token, error))
-                        return FALSE;
+                if (!add_curl_header(&headers, token, error)) {
+                        ret = FALSE;
+                        goto clean;
+                }
 
         if (jsonRequestBody &&
-            !add_curl_header(&headers, "Content-Type: application/json;charset=UTF-8", error))
-                return FALSE;
+            !add_curl_header(&headers, "Content-Type: application/json;charset=UTF-8", error)) {
+                ret = FALSE;
+                goto clean;
+        }
 
         curl_easy_setopt(curl, CURLOPT_HTTPHEADER, headers);
 
         // perform request
         CURLcode res = curl_easy_perform(curl);
         curl_easy_getinfo(curl, CURLINFO_RESPONSE_CODE, &http_code);
-        if (res == CURLE_OK && http_code == 200) {
-                if (jsonResponseParser && fetch_buffer->size > 0) {
-                        JsonParser *parser = json_parser_new_immutable();
-                        if (json_parser_load_from_data(parser, fetch_buffer->payload, fetch_buffer->size, error)) {
-                                JsonNode *resp_root = json_parser_get_root(parser);
-                                g_autofree gchar *str = json_to_string(resp_root, TRUE);
-                                g_debug("Response body: %s", str);
-                                *jsonResponseParser = parser;
-
-                        } else {
-                                g_object_unref(parser);
-                                g_debug("Failed to parse JSON response body. status: %ld", http_code);
-                        }
-                }
-        }
+        curl_slist_free_all(headers);
         if (res != CURLE_OK) {
                 g_set_error(error, RHU_HAWKBIT_CLIENT_CURL_ERROR, res, "%s",
                             curl_easy_strerror(res));
+                ret = FALSE;
+                goto clean;
         } else if (http_code != 200) {
                 g_set_error(error, RHU_HAWKBIT_CLIENT_HTTP_ERROR, http_code,
                             "HTTP request failed: %ld; server response: %s", http_code,
                             fetch_buffer->payload);
+                ret = FALSE;
+                goto clean;
         }
 
+        if (jsonResponseParser && fetch_buffer->size > 0) {
```

Comment:
this change seems to belong logically to the rework of error code handling some commits ago, not to return value or cleanup handling.

Reply:
Right, changing the return type and refactoring of rest_request() got mixed up. Will fix it.